Makefile

```diff
@@ -1,3 +1,45 @@
+
+ifeq (n,$(findstring n,$(firstword -$(MAKEFLAGS))))
+DRY_RUN:=1
+else
+DRY_RUN:=
+endif
+
+ifneq ($(VG),)
+VALGRIND=$(VG)
+endif
+
+ifeq ($(VALGRIND),1)
+override DEBUG ?= 1
+endif
+
+ifneq ($(SAN),)
+override DEBUG ?= 1
+export ASAN_OPTIONS=detect_odr_violation=0
+
+ifeq ($(SAN),mem)
+override SAN=memory
+else ifeq ($(SAN),addr)
+override SAN=address
+endif
+
+ifeq ($(SAN),memory)
+CMAKE_SAN=-DUSE_MSAN=ON
+override CTEST_ARGS += --exclude-regex BruteForceTest.sanity_rinsert_1280
+
+else ifeq ($(SAN),address)
+CMAKE_SAN=-DUSE_ASAN=ON
+else ifeq ($(SAN),leak)
+else ifeq ($(SAN),thread)
+else
+$(error SAN=mem|addr|leak|thread)
+endif
+
+export SAN
+endif # SAN
+
+#----------------------------------------------------------------------------------------------
```

Comment:
this is the next phase - we'll have common definitions for all modules (and this will reside in `mk/readies`).
same mess exists in RediSearch - so let's take it as it is now and I'll clean it up soon for everyone.